packages/api/tests/test-endpoints-granules.js

```diff
@@ -325,3 +329,105 @@ test('move a granule', async (t) => {
     });
   });
 });
+
+test('move a file and update metadata', async (t) => {
+  const bucket = process.env.internal;
+  process.env.bucket = bucket;
+  const buckets = {
+    protected: {
+      name: process.env.internal,
+      type: 'protected'
+    },
+    public: {
+      name: randomString(),
+      type: 'public'
+    }
+  };
+
+  process.env.distEndpoint = 'http://example.com/';
+  await aws.s3().putObject({ Bucket: bucket, Key: `${process.env.stackName}/workflows/buckets.json`, Body: JSON.stringify(buckets) }).promise();
+
+  await aws.s3().createBucket({ Bucket: buckets.public.name }).promise();
+  const newGranule = fakeGranuleFactory();
+  const metadata = fs.createReadStream('tests/data/meta.xml');
+
+  newGranule.files = [
+    {
+      bucket,
+      name: `${newGranule.granuleId}.txt`,
+      filepath: `${process.env.stackName}/original_filepath/${newGranule.granuleId}.txt`
+    },
+    {
+      bucket: buckets.public.name,
+      name: `${newGranule.granuleId}.cmr.xml`,
+      filepath: `${process.env.stackName}/original_filepath/${newGranule.granuleId}.cmr.xml`,
+      filename: `s3://${buckets.public.name}/${process.env.stackName}/original_filepath/${newGranule.granuleId}.cmr.xml`
+    }
+  ];
+
+  await g.create(newGranule);
+
+  await Promise.all(newGranule.files.map(async (file) => {
+    if (file.name === `${newGranule.granuleId}.txt`) {
+      aws.s3().putObject({ Bucket: file.bucket, Key: file.filepath, Body: 'test data' }).promise();
+    }
+    else {
+      aws.s3().putObject({ Bucket: file.bucket, Key: file.filepath, Body: metadata }).promise();
+    }
+  }));
+
+  const destinationFilepath = `${process.env.stackName}/moved_granules`;
+  const destinations = [
+    {
+      regex: '.*.txt$',
+      bucket,
+      filepath: destinationFilepath
+    }
+  ];
+
+  const moveEvent = {
+    httpMethod: 'PUT',
+    pathParameters: {
+      granuleName: newGranule.granuleId
+    },
+    body: JSON.stringify({
+      action: 'move',
+      destinations
+    })
+  };
+
+  await testEndpoint(granuleEndpoint, moveEvent, async (response) => {
+    const body = JSON.parse(response.body);
+    t.is(body.status, 'SUCCESS');
+    t.is(body.action, 'move');
+
+    await aws.s3().listObjects({ Bucket: bucket, Prefix: destinationFilepath }).promise().then((list) => {
```

Comment:
since you are using `await` here this can be written in a less congested way. i.e.:

```
const list = await aws.s3().listObjects({ Bucket: bucket, Prefix: destinationFilepath }).promise();
t.is(list.Contents.length, 1);
list.Contents.forEach((item) => {
  t.is(item.Key.indexOf(destinationFilepath), 0);
});
```